modules/trusted-firmware-m/interface/interface.c

```diff
@@ -34,18 +35,61 @@ int32_t tfm_ns_interface_dispatch(veneer_fn fn,
 		return (int32_t)TFM_ERROR_GENERIC;
 	}
 
+#ifndef CONFIG_ARM_NONSECURE_PREEMTIBLE_SECURE_CALLS
 	/*
 	 * Prevent the thread from being preempted, while executing a Secure
 	 * function. This is required to prevent system crashes that could
 	 * occur, if a thead context switch is triggered in the middle of a
 	 * Secure call.
 	 */
 	k_sched_lock();
+#endif
+
+#if defined(CONFIG_FPU_SHARING) && defined(CONFIG_ARM_NONSECURE_PREEMTIBLE_SECURE_CALLS)
```

Comment:
For ease of reading you can reverse the order of these two here (because above you have #ifndef CONFIG_ARM_NONSECURE_PREEMTIBLE_SECURE_CALLS)